### PR TITLE
Add Payment Sheet for Zip

### PR DESCRIPTION
**Pull request by nahirniak-stripe**
## Summary

The PR adds Payment Sheet on iOS for Zip. 

## Motivation

Enable Zip for more users 

## Testing

Manual testing

**Payment Sheet with Zip:**
Simulator Screen Shot - iPhone 14 Pro - 2023-02-23 at 09 27 29

**Zip selected:**
Simulator Screen Shot - iPhone 14 Pro - 2023-02-23 at 09 28 30

**Redirect:**
simulator_screenshot_D4F2F335-F549-45A5-8F99-7A0219D721A7

**Success:**
Simulator Screen Shot - iPhone 14 Pro - 2023-02-23 at 09 30 25

simulator_screenshot_6C0478C6-BF32-46BF-85A8-063C1C2E8A6E

## Changelog

[Added] Payment Sheet for Zip

### CODE REVIEW

**Comment by porter-stripe** [comment]
Nice! Can we add an automated test for this similar to what we have for Cash App: https://github.com/stripe/stripe-ios/blob/master/Example/PaymentSheet%20Example/PaymentSheetUITest/PaymentSheetUITest.swift#L437

**Comment by wooj-stripe** [comment]
> Nice! Can we add an automated test for this similar to what we have for Cash App: https://github.com/stripe/stripe-ios/blob/master/Example/PaymentSheet%20Example/PaymentSheetUITest/PaymentSheetUITest.swift#L437

@porter-stripe - I was expecting that the v6 playground would have cashapp as a payment method, but looks like we are using automatic payment methods in the test -- if @nahirniak-stripe adds a test like this, how do we guarantee that zip pay shows up as a payment method?

Also, we didn't add these tests for mobilepay & revolut pay because it is relying on `FormSpecProviderTest` to test spec deserialization/rendering and `STPPaymentHandlerStubbedMockedFilesTests` for handling the confirmation for payment methods via luxe.  Given that the next action type is redirect_url, i'm not entirely convinced we need a ui test for this payment method.  That being said -- if it's easy to add, then... sure, lets add it, and then also add it to our documentation ( https://trailhead.corp.stripe.com/docs/luxe/add-lpm/ios-paymentsheet )

**Comment by nahirniak-stripe** [comment]
Hey @porter-stripe @wooj-stripe 👋 

I pushed the test! Thanks for the review 🎉

**Review by porter-stripe — APPROVED** [review]

File: Example/PaymentSheet Example/PaymentSheetUITest/PaymentSheetUITest.swift [hunk 1]
```diff
@@ -434,6 +434,37 @@ class PaymentSheetUITest: XCTestCase {
...
+                "automatic_payment_methods": "on",
+                "currency": "AUD",
+                "merchant_country_code": "AU",
+            ]
+        )
+
+        app.buttons["Checkout (Complete)"].tap()
+        let payButton = app.buttons["Pay A$50.99"]
+
+        // Select Cash App
```
Comment (wooj-stripe) [inline Example/PaymentSheet Example/PaymentSheetUITest/PaymentSheetUITest.swift:451]: copy pasta :)

**Pull request merged**